salt/search/elasticsearch.sls

```diff
@@ -0,0 +1,82 @@
+# Adapted from:
+# - https://github.com/elifesciences/builder-base-formula/blob/master/elife/elasticsearch.sls
+
+elasticsearch-repo:
+    pkgrepo.managed:
+        - humanname: Official Elasticsearch PPA
+        - name: deb https://artifacts.elastic.co/packages/5.x/apt stable main
+        - dist: stable
+        - file: /etc/apt/sources.list.d/elasticsearch.list
+        - key_url: https://artifacts.elastic.co/GPG-KEY-elasticsearch
+
+elasticsearch:
+    group:
+        - present
+
+    user:
+        - present
+        - groups:
+            - elasticsearch
+        - require:
+            - group: elasticsearch
+
+    pkg:
+        - installed
+        - require:
+            - pkg: oracle-java8-installer
+            - pkgrepo: elasticsearch-repo
+
+    service:
+        - running
+        - enable: True
+        - require:
```

Comment:
assuming Java was necessary for this, you want a `require` on `oracle-java8-installer` too here, to force the order of installation